shared/js/dataDriver.js

```diff
@@ -360,10 +376,19 @@ loop.DataDriver = function() {
      *  - {Mixed}  value     Stored value for the record.
      */
     _processRecord(key, data) {
-      let dispatchAction;
-      let dispatchExtra = {};
       // XXX akita bug 1274130: Validate data/key/timestamp before processing.
       let [, type, id] = key.match(/^([^!]+)!(.+)$/);
```

Comment:
Alternatively declare `let type, id;` outside of the try and just do the destructured assignment on the `match` where if there's no match, there'll be a `TypeError: key.match(...) is null`